File: apps/mobile/lib/core/network/me_response.dart
/// Respuesta de `GET /api/v1/me` (Spring). Perfil autorizado del usuario.
class MeResponse {
  const MeResponse({
    required this.id,
    required this.email,
    required this.fullName,
    required this.institution,
    required this.roles,
    required this.permissions,
    required this.offlineWindowHours,
    required this.lastOnlineValidation,
  });

  factory MeResponse.fromJson(Map<String, dynamic> json) {
    return MeResponse(
      id: json['id'] as String,
      email: json['email'] as String,
      fullName: json['fullName'] as String,
      institution: InstitutionDto.fromJson(
        json['institution'] as Map<String, dynamic>,
      ),
      roles: (json['roles'] as List<dynamic>).cast<String>(),
      permissions: (json['permissions'] as List<dynamic>).cast<String>(),
      offlineWindowHours: json['offlineWindowHours'] as int,
      lastOnlineValidation: json['lastOnlineValidation'] as String,
    );
  }

  final String id;
  final String email;
  final String fullName;
  final InstitutionDto institution;
  final List<String> roles;
  final List<String> permissions;
  final int offlineWindowHours;
  final String lastOnlineValidation;
}

class InstitutionDto {
  const InstitutionDto({
    required this.id,
    required this.code,
    required this.name,
  });

  factory InstitutionDto.fromJson(Map<String, dynamic> json) {
    return InstitutionDto(
      id: json['id'] as String,
      code: json['code'] as String,
      name: json['name'] as String,
    );
  }

  final String id;
  final String code;
  final String name;
}

File: services/api/src/main/java/com/pai/api/identity/dto/MeResponse.java
package com.pai.api.identity.dto;

import java.util.List;
import java.util.UUID;

public class MeResponse {

    private final UUID id;
    private final String email;
    private final String fullName;
    private final InstitutionDto institution;
    private final List<String> roles;
    private final List<String> permissions;
    private final int offlineWindowHours;
    private final String lastOnlineValidation;

    public MeResponse(
            UUID id,
            String email,
            String fullName,
            InstitutionDto institution,
            List<String> roles,
            List<String> permissions,
            int offlineWindowHours,
            String lastOnlineValidation) {
        this.id = id;
        this.email = email;
        this.fullName = fullName;
        this.institution = institution;
        this.roles = roles;
        this.permissions = permissions;
        this.offlineWindowHours = offlineWindowHours;
        this.lastOnlineValidation = lastOnlineValidation;
    }

    public UUID getId() {
        return id;
    }

    public String getEmail() {
        return email;
    }

    public String getFullName() {
        return fullName;
    }

    public InstitutionDto getInstitution() {
        return institution;
    }

    public List<String> getRoles() {
        return roles;
    }

    public List<String> getPermissions() {
        return permissions;
    }

    public int getOfflineWindowHours() {
        return offlineWindowHours;
    }

    public String getLastOnlineValidation() {
        return lastOnlineValidation;
    }

    public static class InstitutionDto {
        private final UUID id;
        private final String code;
        private final String name;

        public InstitutionDto(UUID id, String code, String name) {
            this.id = id;
            this.code = code;
            this.name = name;
        }

        public UUID getId() {
            return id;
        }

        public String getCode() {
            return code;
        }

        public String getName() {
            return name;
        }
    }
}
